bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py

```diff
@@ -485,87 +487,123 @@ def set_image(self, image):
         """
         self._yolo.set_image(image)
 
+    @abc.abstractmethod
     def get_candidates(self):
         """
-        :return: all found ball candidates
+        :return: all found candidates
         """
-        return self._yolo.get_candidates("ball")
+        raise NotImplementedError
 
     def compute(self):
         """
         Runs the yolo network
         """
         self._yolo.predict()
 
-class YoloGoalpostDetector(CandidateFinder):
+class YoloBallDetector(YoloDetector):
     """
-    A goalpost detector using the yolo neural network.
+    A ball detector using the yolo neural network.
     This layer connects a single YOLO network with multiple candidate finders for the different classes,
-    in this case the goalpost class.
+    in this case the ball class.
     """
     def __init__(self, config, yolo):
-        """
-        Constructor for the YoloGoalpostDetector.
+        super().__init__(config, yolo)
 
-        :param config: The vision config
-        :param yolo: An YoloHandler implementation that runs the yolo network
+    def get_candidates(self):
         """
-        self._config = config
-        self._yolo = yolo
-
-    def set_image(self, image):
+        :return: all found ball candidates
         """
-        Set a image for yolo. This is cached.
+        return self._yolo.get_candidates("ball")
 
-        :param image: current vision image
-        """
-        self._yolo.set_image(image)
+class YoloGoalpostDetector(YoloDetector):
+    """
+    A goalpost detector using the yolo neural network.
+    This layer connects a single YOLO network with multiple candidate finders for the different classes,
+    in this case the goalpost class.
+    """
+    def __init__(self, config, yolo):
+        super().__init__(config, yolo)
 
     def get_candidates(self):
         """
         :return: all found goalpost candidates
         """
         return self._yolo.get_candidates("goalpost")
 
-    def compute(self):
+class YoloGoalpostDetector(YoloDetector):
```

Comment:
Why do we have two yolo goalpost detectors?

Reply:
Yea that bug took me a few hours... I had a print in one and it didn't print it. Drove me mad...